cookbooks/bcpc-hadoop/recipes/yarn_env.rb

```diff
@@ -9,8 +9,9 @@
 if node.run_list.expand(node.chef_environment)
        .recipes.include?('bach_spark::default')
   yarn_env_generated_values[:YARN_USER_CLASSPATH] =
-    '/usr/spark/current/yarn/spark-yarn-shuffle.jar'
+    "/usr/hdp/current/spark2-client/aux/spark-#{node[:bcpc][:hadoop][:distribution][:active_release]}-yarn-shuffle.jar"
```

Comment:
Style: Symbols --> Strings ([FC001](http://www.foodcritic.io/#FC001))

Could also use the bin dir attribute here.